pkg/awsconfig/testing/firewallConfig.go

```diff
@@ -0,0 +1,479 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/networkfirewall"
+	"github.com/aws/aws-sdk-go/service/networkfirewall/networkfirewalliface"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type Client struct {
+	ec2Client      ec2iface.EC2API
+	firewallClient networkfirewalliface.NetworkFirewallAPI
+}
+
+var wait2 = 360 * time.Second
+
+
+func main() {
+	var ec2Client ec2iface.EC2API
+	var firewallClient networkfirewalliface.NetworkFirewallAPI
+
+	region := os.Getenv("REGION")
+	profile := os.Getenv("PROFILE")
+	if profile == ""{
+		fmt.Println("Profile is not provided, will take in ENV")
+		creds := credentials.NewStaticCredentials(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+		
+		ec2Client = ec2.New(session.New(&aws.Config{
+			Region:      &region,
+			Credentials: creds,
+		}))
+
+		firewallClient = networkfirewall.New(session.Must(session.NewSession()), &aws.Config{
+			Region:      &region,
+			Credentials: creds,
+		})
+
+	}else{
+		sess := session.Must(session.NewSessionWithOptions(session.Options{
+			Config: aws.Config{
+				Region:      &region,
+			},
+			Profile: profile,
+		}))
+		if _, err := sess.Config.Credentials.Get(); err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case "NoCredentialProviders":
+					fmt.Println("could not create AWS session: ", err)
+				default:
+					fmt.Println("could not create AWS session: ", err)
+				}
+			}
+		}
+
+		ec2Client = ec2.New(sess)
+		//create firewall client
+		firewallClient = networkfirewall.New(sess)
+	}
+
+	awsClient := Client{
+		ec2Client,
+		firewallClient,
+	}
```

Comment:
I this can all be combined into one statement

Reply:
implemented